src/codegen/llvm.zig

```diff
@@ -8261,6 +8261,7 @@ pub const FuncGen = struct {
         _ = inst;
         const llvm_fn = self.getIntrinsic("llvm.trap", &.{});
         _ = self.builder.buildCall(llvm_fn.globalGetValueType(), llvm_fn, undefined, 0, .Cold, .Auto, "");
+        _ = self.builder.buildUnreachable();
```

Comment:
Oh, sorry about that! That's why I was seeing `Basic Block in function 'trap.main' does not have terminator!`--thought maybe that only happened because I compiled in debug mode. It still gave me the LLVM IR and all so I thought it would be fine.

Reply:
I opened  #14802 to hopefully prevent this from happening again. I haven't diagnosed the failure yet but I suspect it is likely caused by some of the recent changes to the C backend, or perhaps due to a limitation of our C WASI shims.

I'm sorry that happened to you and others, that must have been very confusing.